### PR TITLE
Feature/utilities - Added basic utilities

### PR DESCRIPTION
**Pull request by MrBisquit**
Added basic utilities:
- checkExists - Checks whether or not an object has value.
- isStringValid - Checks whether or not a string is valid.
- isNumberValid - Checks whether or not a number is valid.
- isArrayValid - Checks whether or not an array is valid.
- getValueOrDefault - Returns the value if it's valid and if not returns the defaultValue.
- doesObjectHaveKey - Checks whether or not an object has a key in it.
- isStringLengthValid - Checks the length of the string is bigger than the minimum and smaller than the maximum (And is valid)
These can help because instead of people typing out if statements they can simply import the utilities file and all of the easy to use utilities are in there.
If you would like to see more utilities or any changes please let me know as I will be working on these.

### CODE REVIEW

File: frontend/src/utils/isStringValid.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import checkExists from "./checkExists";
+/**
+ * Checks if a field is a valid string
+ * @param {*} value - The value to be checked
+ * @returns {boolean} - True if the value is a string.
+ */
+export default function isStringValid(
+  value: string | null | undefined
+): boolean {
+  return checkExists(value) && typeof value == "string";
```
Comment (jaybuurdd) [inline frontend/src/utils/isStringValid.ts:10]: think should be `===`

File: frontend/src/utils/isNumberValid.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * 
+ * @param {number} value - The number to check whether or not it's valid.
+ * @returns {boolean} - Returns true if the number is valid.
+ */
+export default function isNumberValid(
+  value: number | null | undefined
+): boolean {
+  return checkExists(value) && typeof value == "number";
```
Comment (jaybuurdd) [inline frontend/src/utils/isNumberValid.ts:11]: think should be `===`

**Comment by hobi9** [comment]
The code submitted looks good but there is the chance we may never use it. In this project everything that's internal is already fully typed by typescript and everything that's external will pass through zod which will validate it and type it for us so we'd never be in a situation in which we have to manually test types

**Review by hobi9 — APPROVED** [review]

**Comment by borhara** [comment]
I'm not entirely sure about the use case for this

**Comment by MrBisquit** [comment]
> I'm not entirely sure about the use case for this

@SHIV5T3R If we don't use it we can always delete it later.

**Pull request closed without merging**